src/components/DateRangePicker.jsx

```diff
@@ -250,7 +250,8 @@ class DateRangePicker extends React.PureComponent {
     //
     // We handle both situations here by using the ` || ` operator to fallback
     // to *event.target** when **relatedTarget** is not provided.
-    if (this.dayPickerContainer.contains(event.relatedTarget || event.target)) return;
+    if (this.dayPickerContainer.contains(event.relatedTarget || event.target)
+      || this.dayPickerContainer.contains(event.target)) return;
```

Comment:
@marcos0x This is potentially doing the same operation twice, I think. I feel like we should follow @ljharb's advice here and change this to `this.dayPickerContainer.contains(event.relatedTarget) || this.dayPickerContainer.contains(event.target))` in order to prevent us from duplicating logic unnecessarily.

Although the current implementation works, it means that the `|| event.target` has become redundant. If that call fails, it'll revert to checking `event.target` anyway.

Is this something that you would be open to changing?

Reply:
(note that we'll probably need something more complex than an inline `||` here, since we won't want to call `contains` with a falsy element)